src/main/java/pl/javastart/restoffers/category/CategoryController.java

```diff
@@ -0,0 +1,48 @@
+package pl.javastart.restoffers.category;
+
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/api/categories")
+public class CategoryController {
+
+    private final CategoryService categoryService;
+
+    public CategoryController(CategoryService categoryService) {
+        this.categoryService = categoryService;
+    }
+
+    @GetMapping
+    ResponseEntity<List<CategoryDto>> listOfDtoCategories() {
+        if (categoryService.getListOfDtoCategories().isEmpty()) {
+            return ResponseEntity.noContent().build();
+        }
+        return ResponseEntity.ok(categoryService.getListOfDtoCategories());
+    }
+
+    @GetMapping("/names")
+    ResponseEntity<List<String>> listOfCategoryNames() {
+        if (categoryService.listOfCategoryNames().isEmpty()) {
+            return ResponseEntity.notFound().build();
+        }
+        return ResponseEntity.ok(categoryService.listOfCategoryNames());
+    }
+
+    @PostMapping
+    ResponseEntity<CategoryDto> insertCategory(@RequestBody CategoryDto categoryDto) {
+        CategoryDto saved = categoryService.insertCategory(categoryDto);
+        return ResponseEntity.ok(saved);
+    }
```

Comment:
Wysłanie obiektu z ustawionym id, np.
```
{
    "id": 1,
    "name": "abc",
    "description": "Coś dla każdego",
    "offers": 3
}
```
powoduje modyfikację istniejącego obiektu. Nie powinno to być dozwolone. Zakładamy, że POST tworzy obiekt. Do podmiany obiektu powinniśmy raczej wysłać PUT pod adres /api/categories/1